src/agni/server/communication/MessageReceiver.java

```diff
@@ -9,20 +17,196 @@
 import agni.server.receiver.LoginReceiver;
 import agni.server.receiver.UserReceiver;
 
-public class MessageReceiver implements Runnable {
+public class MessageReceiver {
+    private Selector selector = null;
+    private ServerSocketChannel channel = null;
+    private ChannelList channelList = null;
+    
+    ChatReceiver chatReceiver = null;
+    FileReceiver fileReceiver = null;
+    HeartbeatReceiver heartBeatReceiver = null;
+    InfoRequestReceiver infoRequestReceiver = null;
+    LoginReceiver loginReceiver = null;
+    UserReceiver userReceiver = null;
```

Comment:
I don't think these need to be explicitly instantiated to null. Just using 
`ChatReceiver chatReceiver;` I think would be better because these values are never supposed to be null, setting them anywhere like that could be misleading.